### PR TITLE
Show progress bar for upload/download

### PR DESCRIPTION
**Pull request by gregorjerse**

### CODE REVIEW

File: src/resdk/resolwe.py [hunk 1]
```diff
@@ -495,7 +503,20 @@ def _download_files(self, files, download_dir=None):
...
+                    sizes[file_directory] = {
+                        entry["name"]: entry["size"]
+                        for entry in json.loads(
+                            self.session.get(file_directory, auth=self.auth).content
+                        )
+                        if entry["type"] == "file"
+                    }
+                file_size = sizes[file_directory][file_name]
+
+                with tqdm.tqdm(
```
Comment (JureZmrzlikar) [inline src/resdk/resolwe.py:517]: Would be nice to show the name of the file that is being downloaded. Use:
`tqdm(... desc=f"Downloading file {file_name}")`
or something similar.

I realize that there are some `logger.info` calls, but AFAIK, most of the users does not turn them on.

File: src/resdk/uploader.py [hunk 2]
```diff
@@ -143,12 +160,19 @@ def _upload_s3(self, file_path: Union[Path, str]) -> str:
         bucket_name = self.upload_config["config"]["bucket_name"]
 
         destination = Path(prefix) / str(uuid.uuid4())
-        self._s3_client.upload_file(
-            Filename=str(file_path), Key=destination.as_posix(), Bucket=bucket_name
-        )
+        # Use progress bar to show progress.
+        with tqdm.tqdm(
+            total=Path(file_path).stat().st_size, disable=not show_progress
```
Comment (JureZmrzlikar) [inline src/resdk/uploader.py:165]: Add description, same as for download.

File: src/resdk/uploader.py [hunk 3]
```diff
@@ -169,7 +193,9 @@ def _upload_local(self, file_path: Union[Path, str]):
         if self.resolwe.session is None:
             raise RuntimeError("Session has not been initialized.")
 
-        with file_path.open("rb") as file_:
+        with file_path.open("rb") as file_, tqdm.tqdm(
+            total=file_size, disable=not show_progress
```
Comment (JureZmrzlikar) [inline src/resdk/uploader.py:197]: Add description, same as for download.

**Review by JureZmrzlikar — COMMENTED** [review]
Looks good, just a few minor comments.

File: pyproject.toml [hunk 4]
```diff
@@ -75,6 +75,8 @@ test = [
     "mock",
     "pydocstyle[toml]",
     "pytest-cov",
+    "types-tqdm",
```
[the remark this answers is not in the document]
Comment (gregorjerse) [reply pyproject.toml:78]: No idea. In my IDE they all look the same.

File: src/resdk/uploader.py (hunk 2 above)
Comment (gregorjerse) [reply src/resdk/uploader.py:165]: Done.

File: src/resdk/resolwe.py (hunk 1 above)
Comment (JureZmrzlikar) [reply src/resdk/resolwe.py:517]: Where?

File: src/resdk/uploader.py (hunk 3 above)
Comment (JureZmrzlikar) [reply src/resdk/uploader.py:197]: Where?

**Review by JureZmrzlikar — COMMENTED** [review]
Please also fix tests when you find time.

**Review by JureZmrzlikar — APPROVED** [review]
Works fine, looks good. Thanks for your effort!

**Pull request merged**